network/config.go

```diff
@@ -61,12 +62,34 @@ func DefaultConfig() *Config {
 	}
 }
 
+func validateAddress(address []string) bool {
+	for _, addr := range address {
+		isValid := isValidAddress(addr)
+		if !isValid {
+			return false
+		}
+	}
+	return true
+}
+
+func isValidAddress(address string) bool {
```

Comment:
Why do we return `bool` and not `error`? An error message would give users some information about why the address is invalid.
This function can be merged with `validateAddresses`.